Make GraphicsLayerTextureMapper code platform agnostic

donny-dont opened the pull request:
#### 0b29e27aecb105dbba819c2ae00b71eab39a098e
<pre>
Make GraphicsLayerTextureMapper code platform agnostic
<a href="https://bugs.webkit.org/show_bug.cgi?id=246497">https://bugs.webkit.org/show_bug.cgi?id=246497</a>

Reviewed by Fujii Hironori.

Replaces `PLATFORM(WIN)` code with checks for
`USE(GRAPHICS_LAYER_TEXTURE_MAPPER)` or `USE(GRAPHICS_LAYER_WC)` where
applicable. Modify types to platform specific ones. The patch largely
modifies code from 225218@main to be port agnostic.

* Source/WebKit/Shared/WebPageCreationParameters.cpp:
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostTextureMapper.cpp:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostTextureMapper.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebPage/WebPage.h:

Canonical link: <a href="https://commits.webkit.org/255582@main">https://commits.webkit.org/255582@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/dfe978dbd42ab42a9fe58e6fe6541cc4f2f7ead3

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92919 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/2126 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/23499 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/102639 "Built successfully") | 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/2131 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/30462 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/85311 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/98769 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/98585 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/1458 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/79399 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/28372 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/83371 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/83099 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/71491 "Found 1 new API test failure: /TestWTF:WTF_WordLock.ManyContendedLongSections (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36862 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/16996 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/34666 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/18186 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3868 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38536 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/40788 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/40457 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/37366 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebKit/Shared/WebPageCreationParameters.h:
```diff
@@ -187,7 +187,7 @@ struct WebPageCreationParameters {
 #endif
-#if PLATFORM(WIN)
+#if USE(GRAPHICS_LAYER_TEXTURE_MAPPER)
     uint64_t nativeWindowHandle;
```
nativeWindowHandle is not only for GraphicsLayerTextureMapper, but also for GraphicsLayerWC. And, I have a plan to remove GraphicsLayerTextureMapper after removing WinCairo WK1 and migrating Microsoft Playwright to GraphicsLayerWC.
I think it should be `#if USE(GRAPHICS_LAYER_TEXTURE_MAPPER) || USE(GRAPHICS_LAYER_WC)`.

donny-dont replied:
Ok updated @fujii

donny-dont committed:
Make GraphicsLayerTextureMapper code platform agnostic
https://bugs.webkit.org/show_bug.cgi?id=246497

Reviewed by Fujii Hironori.

Replaces `PLATFORM(WIN)` code with checks for
`USE(GRAPHICS_LAYER_TEXTURE_MAPPER)` or `USE(GRAPHICS_LAYER_WC)` where
applicable. Modify types to platform specific ones. The patch largely
modifies code from 225218@main to be port agnostic.

* Source/WebKit/Shared/WebPageCreationParameters.cpp:
* Source/WebKit/Shared/WebPageCreationParameters.h:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostTextureMapper.cpp:
* Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/LayerTreeHostTextureMapper.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/WebPage/WebPage.h:

Canonical link: https://commits.webkit.org/255582@main
